docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+This library ensures that transformations affect the image and update the bounding boxes accordingly.
+The 🤗 Datasets library documentation has a detailed [guide on how to augment images for object detection](https://huggingface.co/docs/datasets/object_detection),
+and it uses the exact same dataset as an example. Apply the same approach here, i.e., resize each image to (480, 480),
+flip it horizontally, and brighten it:
+
+```py
+>>> import albumentations as A
+>>> import numpy as np
+>>> import torch
+
+>>> transform = A.Compose(
+...     [
+...         A.Resize(480, 480),
+...         A.HorizontalFlip(p=1.0),
+...         A.RandomBrightnessContrast(p=1.0),
+...     ],
+...     bbox_params=A.BboxParams(format="coco", label_fields=["category"]),
+... )
+```
+
+`DetrImageProcessor` expects the annotations to be in the following format: `{‘image_id’: int, ‘annotations’: List[Dict]}`,
+ where each `Dict` is a COCO object annotation. Building up gradually, add a function to reformat annotations for a single example:
+
+```py
+>>> def formatted_anns(image_id, category, area, bbox):
+
+...     annotations = []
+...     for i in range(0, len(category)):
+...         new_ann = {
+...             "image_id": image_id,
+...             "category_id": category[i],
+...             "isCrowd": 0,
+...             "area": area[i],
+...             "bbox": list(bbox[i]),
+...         }
+...         annotations.append(new_ann)
+
+...     return annotations
+```
+
+Now you can combine the image and annotation transformations to use on a batch of examples:
+
+```py
+>>> # transforming a batch
+>>> def transform_aug_ann(examples):
+...     image_ids = examples["image_id"]
+...     images, bboxes, area, categories = [], [], [], []
+...     for image, objects in zip(examples["image"], examples["objects"]):
+...         image = np.array(image.convert("RGB"))[:, :, ::-1]
+...         out = transform(image=image, bboxes=objects["bbox"], category=objects["category"])
+
+...         area.append(objects["area"])
+...         images.append(torch.tensor(out["image"]).flip(-1).permute(2, 0, 1))
+...         bboxes.append(out["bboxes"])
+...         categories.append(out["category"])
+
+...     targets = [
+...         {"image_id": id_, "annotations": formatted_anns(id_, cat_, ar_, box_)}
+...         for id_, cat_, ar_, box_ in zip(image_ids, categories, area, bboxes)
+...     ]
+
+...     return image_processor(images=images, annotations=targets, return_tensors="pt")
+```
+
+Apply this preprocessing function to the entire dataset using 🤗 Datasets `with_transform` method. This method applies
+transformations on the fly when you load an element of the dataset.
+
+At this point, you can check what an example from the dataset looks like after the transformations. You should see a tensor
+with `pixel_values`, a tensor with `pixel_mask`, and `labels`.
+
+```py
+>>> cppe5["train"] = cppe5["train"].with_transform(transform_aug_ann)
+>>> cppe5["train"][15]
```

Comment:
Feel free to also append the output in a trimmed manner here (c.f. https://moon-ci-docs.huggingface.co/docs/transformers/pr_20925/en/tasks/semantic_segmentation).

Reply:
Might also be worth visualizing how augmented images with bounding boxes look like. 

We might want to reuse the visualization code here. Maybe turn it into a function so that the overall code stays cleaner?

Reply:
I'll add the output, but I don't think it is necessary to add the visualization. This guide is already quite large, and there's a separate doc page about this exact augmentation on this exact dataset that has all of the visualizations. I think it's enough to link it here.

Reply:
Fair enough. But one/two viz won't hurt I guess. I will let you choose those.